src/common/utils/debouncePromise.ts

```diff
@@ -4,6 +4,6 @@ export function debouncePromise<T, F extends any[]>(func: (...args: F) => T | Pr
   let timer: NodeJS.Timeout;
   return (...params: any[]) => new Promise((resolve, reject) => {
     clearTimeout(timer);
-    timer = setTimeout(() => resolve(func.apply(this, params)), timeout);
+    timer = global.setTimeout(() => resolve(func.apply(this, params)), timeout);
```

Comment:
Why `global.setTimeout`? Any difference?

Reply:
Typescript complained about this:
> src/common/utils/debouncePromise.ts:7:5 - error TS2322: Type 'number' is not assignable to type 'Timeout'.

Basically typescript thinks it runs in a browser (`lib.DOM` is enabled) unless we explicitly say to use nodejs function.